catalog/task/run-file-updates/0.2/run-file-updates.yaml

```diff
@@ -0,0 +1,65 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: run-file-updates
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to run file updates
+  params:
+    - name: jsonKey
+      type: string
+      description: The json key containing the file updates
+      default: ".spec.extraData.fileUpdates[]"
+    - name: fileUpdatesPath
+      type: string
+      description: The path to the file containing the file updates
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+      default: "snapshot_spec.json"
+    - name: request
+      type: string
+      description: Name of the request
+      default: "file-updates"
+    - name: synchronously
+      type: string
+      description: Whether to run synchronously or not
+      default: "true"
+  workspaces:
+    - name: data
+      description: Workspace where the file updates to apply are defined
+  steps:
+    - name: run-script
+      image: quay.io/hacbs-release/release-utils:a918b3a7cbd40add5955a2f1acb4c3dbb29c44e9
+      script: |
+        #!/bin/sh
+        set -ex
+
+        # Read JSON file into a variable
+        jsonData=$(cat $(workspaces.data.path)/$(params.fileUpdatesPath))
+
+        # Extract the key from the JSON file
+        fileUpdates=$(echo "${jsonData}" | jq -rc "$(params.jsonKey)")
+  
+        # Iterate over the extracted array and call the script
+        IFS=$'\n'
```

Comment:
This won't be needed if you iterate over indexes. But if you insisted on doing it this way, this variable should only be defined for the `for` command below.

Reply:
fixed.

Reply:
Ah, sorry, I didn't realize that. In that case I would be fine to keep it intact. Up to you.